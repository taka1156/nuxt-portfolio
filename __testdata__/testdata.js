const MAX_NAV_DATA = 4;
const MAX_PROFILE_DATA = 4;
const MAX_CARD_DATA = 6;
const dummyImgUrl = 'http://placehold.jp/150x150.png';
const GITHUB_URL = 'https://github.com/taka1156/audio-player';
const WEBSITE_URL = 'https://takatechblog.netlify.app/';

// factory
const dummyFactory = (num, fn) => {
  return [...new Array(num)].map((_, i) => fn(i + 1));
};

// template
const dummyNavTemplate = () => ({
  name: 'ダミーナビゲーション',
  to: '/test1',
  img: dummyImgUrl,
});

const dummyCardTemplate = () => ({
  title: 'ダミーカード',
  img: { url: dummyImgUrl },
  content2:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
});

const dummyCardLinkTemplate = i => ({
  title: 'ダミーカード',
  img: { url: dummyImgUrl },
  content2:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  link: i % 2 === 0 ? GITHUB_URL : WEBSITE_URL,
});

const dummyProfileTemplate = () => {
  return 'ダミープロフ';
};

const dummyImg = {
  imgUrl: dummyImgUrl,
  imgAlt: 'ダミーの画像',
};

const dummyRoutes = dummyFactory(MAX_NAV_DATA, () => dummyNavTemplate());

const dummyCards = dummyFactory(MAX_CARD_DATA, () => dummyCardTemplate());

const dummyCardsLink = dummyFactory(MAX_CARD_DATA, i => dummyCardLinkTemplate(i));

const dummyProfiles = dummyFactory(MAX_PROFILE_DATA, () => dummyProfileTemplate());

const dummyLogo = 'ダミーロゴ';

export {
  dummyImg,
  dummyLogo,
  dummyRoutes,
  dummyCards,
  dummyCardsLink,
  dummyProfiles,
};
