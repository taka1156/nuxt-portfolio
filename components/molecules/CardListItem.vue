<template>
  <div>
    <article class="contents" @click="jump(cardInfo.link ? cardInfo.link : '')">
      <card-title :card-title="cardInfo.title" />
      <figure>
        <card-img :card-img="fomatImg(cardInfo.img.url)" />
        <figcaption>
          <card-text :card-text="cardInfo.content1" />
          <card-text :card-text="cardInfo.content2" />
        </figcaption>
      </figure>
    </article>
  </div>
</template>

<script>
import CardTitle from '../atoms/CardTitle';
import CardImg from '../atoms/CardImg';
import CardText from '../atoms/CardText';

export default {
  name: 'CardItem',
  components: {
    'card-title': CardTitle,
    'card-img': CardImg,
    'card-text': CardText,
  },
  props: {
    cardInfo: {
      type: Object,
      default: () => {},
      required: true,
    },
  },
  methods: {
    fomatImg(img) {
      return `${img}?fit=fillmax&fill-color=gray&w=388&h=312`;
    },
    jump(link) {
      if (link !== '') {
        location.href = link;
      }
    },
  },
};
</script>

<style scoped>
.contents {
  width: 100%;
  height: 100%;
  margin: 2vh auto;
}
</style>
