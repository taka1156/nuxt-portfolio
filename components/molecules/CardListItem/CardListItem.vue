<template>
  <div>
    <article class="card-list-item">
      <figure>
        <base-img-lazy
          :img-url="card.img.url"
          :img-alt="`${card.title}の画像`"
          :size="`${card.link == null ? 'sm' : 'lg'}`"
        />
        <div class="card-list-item__border" />
        <figcaption class="card-list-item__caption">
          <base-title>{{ card.title }}</base-title>
          <base-text>{{ card.content2 }}</base-text>
        </figcaption>
      </figure>
      <base-link v-if="card.link != null" :route-to="card.link" :is-target="true">
        {{ isGithubRepo(card) ? 'GitHub' : 'WebSite' }}
      </base-link>
    </article>
  </div>
</template>

<script>
import BaseTitle from '../../atoms/BaseTitle/BaseTitle';
import BaseText from '../../atoms/BaseText/BaseText';
import BaseLink from '../../atoms/BaseLink/BaseLink';
import BaseImgLazy from '../../atoms/BaseImgLazy/BaseImgLazy';

export default {
  name: 'CardListItem',
  components: {
    'base-title': BaseTitle,
    'base-text': BaseText,
    'base-link': BaseLink,
    'base-img-lazy': BaseImgLazy
  },
  props: {
    /**
     * カードリスト１つ分
     */
    card: {
      type: Object,
      required: true
    }
  },
  methods: {
    isGithubRepo({ link }) {
      /**
       * githubリポジトリかどうかを判定する処理
       */
      return `${link}`.includes('github') && !`${link}`.includes('github.io');
    }
  }
};
</script>

<style scoped>
/* css reset */
figure {
  padding: 0;
  margin: 0;
}

figcaption {
  padding: 0;
  margin: 0;
}

/* css reset */
::v-deep .base-text--extend {
  box-sizing: border-box;
  display: block;
  width: 100%;
  height: 100px;
  margin: 0 auto;
  text-align: left;
  word-wrap: break-word;
}

::v-deep .base-link--extend {
  display: block;
  width: 75%;
  padding: 10px;
  margin: 10px auto;
  font-size: 1em;
  color: gray;
  background-color: white;
  border: 1px solid gray;
  border-radius: 5px;
  outline: none;
}

::v-deep .base-link--extend:active,
::v-deep .base-link--extend:hover,
::v-deep .base-link--extend:focus {
  color: white;
  background-color: gray;
}

::v-deep .base-img--extend {
  padding: 3px;
  margin: 0 auto;
}

.card-list-item {
  width: 100%;
  height: 100%;
  margin: 0 auto;
  border: solid 0.6px rgb(211, 211, 211);
  box-shadow: 5px 10px 20px rgba(0, 0, 0, 0.25);
}

.card-list-item__border {
  border-bottom: 1px dashed black;
}

.card-list-item__caption {
  padding: 10px;
}
</style>
