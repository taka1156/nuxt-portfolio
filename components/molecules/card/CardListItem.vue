<template>
  <div>
    <article class="card-list-item">
      <figure>
        <card-img
          :card-img="cardInfo.img.url"
          :img-alt="`${cardInfo.title}の画像`"
          :is-portfolio="cardInfo.link != null"
        />
        <div class="card-list-item__border" />
        <figcaption>
          <card-title>{{ cardInfo.title }}</card-title>
          <base-text class="base-text--card">{{ cardInfo.content2 }}</base-text>
        </figcaption>
      </figure>
      <base-btn v-if="cardInfo.link != null" @click="$emit('jump', cardInfo)">
        {{ isGithubRepo(cardInfo) ? 'GitHub' : 'WebSite' }}
      </base-btn>
    </article>
  </div>
</template>

<script>
import BaseText from '../../atoms/base/BaseText';
import BaseBtn from '../../atoms/base/BaseBtn';
import CardTitle from '../../atoms/card/CardTitle';
import CardImg from '../../atoms/card/CardImg';

export default {
  name: 'CardListItem',
  components: {
    'card-title': CardTitle,
    'card-img': CardImg,
    'base-text': BaseText,
    'base-btn': BaseBtn,
  },
  props: {
    cardInfo: {
      type: Object,
      default: () => {},
      required: true,
    },
  },
  methods: {
    isGithubRepo({ link }) {
      // githubがリンクに含まれてる、かつgithub.ioではないならリポジトリなので表示変更
      return `${link}`.includes('github') && !`${link}`.includes('github.io');
    },
  },
};
</script>

<style scoped>
/* css reset */
figure {
  margin: 0;
  padding: 0;
}

figcaption {
  margin: 0;
  padding: 0;
}

/* css reset */

.card-list-item {
  height: 100%;
  width: 100%;
  margin: 0 auto;
  border: solid 0.6px rgb(211, 211, 211);
  box-shadow: 5px 10px 20px rgba(0, 0, 0, 0.25);
}

.card-list-item__border {
  border-bottom: 1px dashed black;
}
</style>
