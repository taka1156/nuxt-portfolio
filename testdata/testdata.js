const dummyLogo = 'ダミー';

const dummyRoutes = [
  {
    name: 'nav-item',
    to: '/test1',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test2',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test3',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test4',
    img: 'http://placehold.jp/150x150.png',
  },
];

const dummyProfile = [
  'TypeScriptの学習',
  'Laravel+Vueでサービスを作る',
  'AWSなどインフラ周りの学習',
  'CSS/SASSの学習',
];

const dummyCards = [
  {
    title: 'PHP',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '1~4ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'Laravel',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '2~3ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'JavaScript(ES6)',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '６ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'Vue.js',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '8ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
];

const dummyCardsLink = [
  {
    title: 'ブログサイト',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Nuxt.js(Vue.js), axios, microCMS, BootStrap4, vue-infinite-loading',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://takatechblog.netlify.app/',
  },
  {
    title: 'マークダウンエディター',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Vue.js, Electron, BootStrap4',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/markdown-editor',
  },
  {
    title: 'LineBot',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Express.js, axios, @line/bot-sdk , Vercel(Zeit Now)',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/line-bot',
  },
  {
    title: 'オーディオプレイヤー',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Vue.js, Electron, JSMediaTags',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/audio-player',
  },
];

export { dummyLogo, dummyRoutes, dummyProfile, dummyCards, dummyCardsLink };
