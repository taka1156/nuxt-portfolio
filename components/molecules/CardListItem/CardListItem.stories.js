import CardListItem from './CardListItem.vue';
import { dummyCards, dummyCardsLink } from '@/__testdata__/testdata.js';

export default {
  title: 'Molecules/CardListItem',
  component: CardListItem
};

const Template1 = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { CardListItem },
  template: '<card-list-item v-bind="$props" />'
});

const Template2 = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { CardListItem },
  template: '<card-list-item v-bind="$props" />'
});

export const Default = Template1.bind({});
Default.args = { card: dummyCards[0] };

export const Link = Template2.bind({});
Link.args = { card: dummyCardsLink[0] };
