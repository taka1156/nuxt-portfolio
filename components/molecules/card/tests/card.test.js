import { mount } from '@vue/test-utils';
import CardListItem from '../CardListItem.vue';
import { dummyCards, dummyCardsLink } from '@/__testdata__/testdata.js';

// CardListItem
describe('CardListItem', () => {
  const cardListItem = propsData => {
    return mount(CardListItem, {
      propsData: {
        ...propsData,
      },
    });
  };

  it('CardListItemの初期値(linkなし): dummyCards', () => {
    const wrapper = cardListItem({ card: dummyCards[0] });
    console.log(wrapper.vm.$options.props.card.default());
    expect(wrapper.vm.$options.props.card.required).toBe(true);
    // linkがない時はボタンがないことを確認
    const btnTag = wrapper.find('button');
    expect(btnTag.exists()).toBe(false);
    expect(wrapper.vm.card).toBe(dummyCards[0]);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('CardListItemの初期値(linkあり): dummyCardsLink', () => {
    const wrapper = cardListItem({ card: dummyCardsLink[0] });
    expect(wrapper.vm.$options.props.card.required).toBe(true);
    // linkがある時はボタンがあることを確認
    const btnTag = wrapper.find('button');
    expect(btnTag.text()).toBe('WebSite');
    expect(wrapper.vm.card).toBe(dummyCardsLink[0]);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('isGithubRepoメソッド: dummyCardsLink', () => {
    // 空だとエラーを吐くのでデータを入れる
    const wrapper = cardListItem({ card: dummyCardsLink[0] });
    // githubのリポジトリとデモサイトの時の比較
    expect(wrapper.vm.isGithubRepo(dummyCardsLink[0])).toBe(false);
    expect(wrapper.vm.isGithubRepo(dummyCardsLink[1])).toBe(true);
  });
});
