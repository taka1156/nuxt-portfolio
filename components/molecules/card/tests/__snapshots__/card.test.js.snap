// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardListItem CardListItemの初期値(linkあり): dummyCardsLink 1`] = `
"<div>
  <article class=\\"card-list-item\\">
    <figure><img alt=\\"ブログサイトの画像\\" class=\\"card-img card-img--lg\\">
      <div class=\\"card-list-item__border\\"></div>
      <figcaption>
        <h2 class=\\"card-title\\">ブログサイト</h2>
        <p class=\\"base-text base-text--card\\">この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい</p>
      </figcaption>
    </figure> <button class=\\"btn\\">
      WebSite
    </button>
  </article>
</div>"
`;

exports[`CardListItem CardListItemの初期値(linkなし): dummyCards 1`] = `
"<div>
  <article class=\\"card-list-item\\">
    <figure><img alt=\\"PHPの画像\\" class=\\"card-img card-img--sm\\">
      <div class=\\"card-list-item__border\\"></div>
      <figcaption>
        <h2 class=\\"card-title\\">PHP</h2>
        <p class=\\"base-text base-text--card\\">この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい</p>
      </figcaption>
    </figure>
    <!---->
  </article>
</div>"
`;
